test/FunctionalTests/Tests/DataServices/UnitTests/ClientCSharpUnitTests/DataWebClientCSharp/LinqTests.cs

```diff
@@ -8153,6 +8153,34 @@ public void QueryRowCount_UriBuilder()
             }
         }
 
+        [TestMethod]
+        public void VerifyCountIsNotCarriedOver()
+        {
+            string expectedLongCountFunctionUri = "/service.svc/Leagues/LongCountFunction(Param1='Param1',Param2='Param2')?$orderby=ID";
+            string expectedCountFunctionUri = "/service.svc/Leagues/CountFunction(Param1='Param1',Param2='Param2')?$orderby=ID";
+
+            var countQuery = context.CreateQuery<League>("Leagues");
+            var longcountQuery = context.CreateQuery<League>("Leagues");
+
+            var countFunctionQuery = countQuery.CreateFunctionQuery<League>("CountFunction", true, new UriOperationParameter("Param1", "Param1"),
+                    new UriOperationParameter("Param2", "Param2"));
+            var longCountFunctionQuery = longcountQuery.CreateFunctionQuery<League>("LongCountFunction", true, new UriOperationParameter("Param1", "Param1"),
+                    new UriOperationParameter("Param2", "Param2"));
+            try
+            {
+                //check for both count an longcount
+                countFunctionQuery.Count();
+                longCountFunctionQuery.LongCount();
+
+            }
+            catch(DataServiceQueryException)

```

Comment:
I don't understand this try/catch inside the test. Is it necessary? What would cause the this to fail?

Reply:
The exception is caught because there is not real service being called but the URI is formed. Our interest in this case is the URI constructed which we are able to check within the catch clause.

Reply:
Shouldn't Count and LongCount be lazy-evaluated though? Calling them shouldn't trigger a network request

Reply:
They are evaluated while making the call. They are not evaluated beforehand.